servus/serializable.h

```diff
@@ -77,46 +76,54 @@ class Serializable
      *
      * @return the binary representation of this object.
      */
-    Data toBinary() const { return _toBinary(); }
+    Data toBinary() const;
 
     /**
      * Update this serializable from its JSON representation.
      * @return true on success, false on error.
      */
-    bool fromJSON( const std::string& json ) { return _fromJSON( json ); }
+    bool fromJSON( const std::string& json );
 
     /** @return the JSON representation of this serializable. */
-    std::string toJSON() const { return _toJSON(); }
+    std::string toJSON() const;
     //@}
 
     /** @name Change Notifications */
     //@{
     /** Function for change notification. */
 #ifdef SERVUS_USE_CXX03
     typedef boost::function< void() > ChangeFunc;
+    typedef boost::function< void() > DeserializeFunc;
+    typedef boost::function< void() > SerializeFunc;
 #else
     typedef std::function< void() > ChangeFunc;
+    typedef std::function< void() > DeserializeFunc;
+    typedef std::function< void() > SerializeFunc;
 #endif
 
     /**
-     * Set a new function called after the object has been updated.
-     *
-     * @return the previously set function.
+     * Register a new function called after after every local change on this
+     * instance (e.g. Collage distributable). To deregister the callback,
+     * call this function with an empty function.
      */
-    SERVUS_API ChangeFunc setUpdatedFunction( const ChangeFunc& func );
+    SERVUS_API void registerChangedCallback( const ChangeFunc& func );
 
     /**
-     * Set a new function called when a request has been received.
-     *
-     * Invoked before the object is published.
-     * @return the previously set function.
+     * Register a new function called after the object has been updated remotely
+     * (via a subscriber, a http server, loading from file...). To deregister
+     * the callback, call this function with an empty function.
+     */
+    SERVUS_API void registerDeserializedCallback( const DeserializeFunc& func );
+
+    /**
+     * Register a new function called when a request has been received.
+     * Invoked before the object is serialized. To deregister the callback,
+     * call this function with an empty function.
      */
-    SERVUS_API ChangeFunc setRequestedFunction( const ChangeFunc& func );
+    SERVUS_API void registerSerializeCallback( const SerializeFunc& func );
 
-    /** @internal used by ZeroEQ to invoke updated function */
-    SERVUS_API void notifyUpdated() const;
-    /** @internal used by ZeroEQ to invoke updated function */
-    SERVUS_API void notifyRequested() const;
+    /** @internal used by ZeroEQ to invoke 'changed' function */
+    SERVUS_API void notifyChanged() const;
```

Comment:
Right, I see. Let me try to put everything together from the different PRs:
The ChangedCallback is only used by Collage Distributable, which is a child class of Serializable. A callback would not even be needed at the moment. A simpler alternative could be:
Servus:
private virtual void _onChanged() {}
ZeroBuf:
public void notifyChanged() { _onChanged(); }
Collage:
private void _onChanged() final { _dirty = true; }